### PR TITLE
Update glutin to v0.23.0

### LINKED ISSUE #1359: Ctrl+q not working
Which operating system does the issue occur on?
macOS

`Ctrl+q` is not working unless I add this line to `key_bindings`:

```yml
- { key: Q, mods: Control, chars: "\x11" }
```

Other `Ctrl` keys seem working with default `key_bindings` config.

### LINKED ISSUE #1465: Proper support for (un)hinding on macOS
After testing the latest changes in dbcb5885adb1f0b23c29838ef8dd837212322409 (#1281) I noticed, that I can't regain focus, or rather partially, of the Alacritty window after using the `CMD + H` keystroke.

~The first part shows the normal behavior and the second part, after the key stroke was pressed. You can also see, that the window action buttons also don't work anymore (keystrokes are still working though).~

### LINKED ISSUE #3150: MacOS minimize action only works with decorations
The `Minimize` keybinding action on macOS only works when window decorations are enabled.

See upstream winit bug: https://github.com/rust-windowing/winit/issues/1361

### LINKED ISSUE #3191: Incorrect mouse position
Using the mouse to select text is off when on Retina screen.

Bug introduced in https://github.com/alacritty/alacritty/commit/c34ec12c309695e4c14d8e50b5f3f54198f70775

### System

OS: macOS
Version: master

Reproduction:

Open Alacritty on retina screen
Try to select text
Selection is off from actual cursor position

### PR DESCRIPTION
**Pull request by kchibisov**
Fixes #3191.
Fixes #3150.
Fixes #1465.
Fixes #1359.

### CODE REVIEW

**Comment by kchibisov** [comment]
I'm not sure why #3177 was linked to a new glutin update, since it's about Dark Theme thing failing, however the upstream issue on #3177 is about Windows 7 and 8 builds.

**Comment by jansol** [comment]
Confirming that this makes hide/unhide work as expected on macOS.

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,6 +46,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Backspace deleting characters while IME is open on macOS
 - Handling of OpenType variable fonts
 - Expansion of block-selection on partially selected full-width glyphs
+- Minimize action only works with decorations on macOS
+- Hiding the window behavior on macOS
```
Comment (chrisduerr) [inline CHANGELOG.md]:
```suggestion
- Window permanently vanishing after hiding on macOS
```

We're not 'hiding the window behavior', that sounds a bit strange.

File: CHANGELOG.md [hunk 2]
```diff
@@ -46,6 +46,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Ctrl + q not working on macOS
```
Comment (chrisduerr) [inline CHANGELOG.md]: This does not only apply to ctrl + q as far as I understand.

```suggestion
- Modifier key combinations like Ctrl + Q not generating characters on macOS
```

File: alacritty/src/window.rs [hunk 3]
```diff
@@ -176,7 +176,6 @@ impl Window {
         self.window().scale_factor()
     }
 
-    #[inline]
```
[the remark this answers is not in the document]
Comment (kchibisov) [reply alacritty/src/window.rs]: pressed `dd` at some point I guess, shrug.

**Pull request merged**